fix documentation on tryCatch method

edjroz opened the pull request:
In regards #2519. 

However I can see that the it was fixed on the source with another example 
`R.tryCatch(JSON.parse, R.F)({});      //=> false`

 But hasn't been built to reflect it on the documentation

CrossEye commented on source/tryCatch.js:
```diff
@@ -22,8 +21,9 @@ import _curry2 from './internal/_curry2';
- *      R.tryCatch(JSON.parse, R.F)({});      //=> false
- */
+ *      R.tryCatch(R.test(/^a/, R.__), R.F)('b'); // => false
+ *      R.tryCatch(R.times(R.identity), R.always([]))('s') // => []
```
I think perhaps we should add a much more explicit example:

   R.tryCatch(() => {throw 'oops'}, () => 'foo')('bar')  //=> 'foo'

edjroz commented:
Changed the examples (I squashed the commits to avoid pollution) 

for a simple scenario:
```Javascript
R.tryCatch(() => { throw 'foo'}, R.always('catched'))('bar') // => 'catched'
```
and a little more advanced one:
```Javascript
R.tryCatch(R.times(R.identity), R.always([]))('s') // => []
```

CrossEye commented:
Thank you!

JulianoGTZ commented:
Is this issue still open?

CrossEye commented:
This was merged some time ago; it is closed.
